tests/integration/scaling_helpers.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import re
+from pathlib import Path
+from subprocess import PIPE, check_output
+from typing import Dict
+
+import yaml
+from kazoo.client import KazooClient
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+APP_NAME = METADATA["name"]
+
+
+def get_password(model_full_name: str) -> str:
+    # getting relation data
+    show_unit = check_output(
+        f"JUJU_MODEL={model_full_name} juju show-unit {APP_NAME}/0",
+        stderr=PIPE,
+        shell=True,
+        universal_newlines=True,
+    )
+    response = yaml.safe_load(show_unit)
+    password = response[f"{APP_NAME}/0"]["relation-info"][0]["application-data"]["super_password"]
+    return password
+
+
+def restart_unit(model_full_name: str, unit: str) -> None:
+    # getting juju id
+    machine_id = check_output(
+        f"JUJU_MODEL={model_full_name} juju status | grep {unit} | awk '{{ print $4 }}'",
+        stderr=PIPE,
+        shell=True,
+        universal_newlines=True,
+    )
+
+    # getting lxc machine name
+    machine_name = check_output(
+        f"JUJU_MODEL={model_full_name} juju machines | grep awk '{{print $4}}' | grep -e '-{machine_id}'| head -1",
+        stderr=PIPE,
+        shell=True,
+        universal_newlines=True,
+    )
+    _ = check_output(
+        f"lxc restart {machine_name}",
+        stderr=PIPE,
+        shell=True,
+        universal_newlines=True,
+    )
+
+
+def write_key(host: str, password: str, username: str = "super") -> None:
+    kc = KazooClient(
+        hosts=host,
+        sasl_options={"mechanism": "DIGEST-MD5", "username": username, "password": password},
+    )
+    kc.start()
+    kc.create_async("/legolas", b"hobbits")
+    kc.stop()
+    kc.close()
+
+
+def check_key(host: str, password: str, username: str = "super") -> None:
+    kc = KazooClient(
+        hosts=host,
+        sasl_options={"mechanism": "DIGEST-MD5", "username": username, "password": password},
+    )
+    kc.start()
+    assert kc.exists_async("/legolas")
+    value, _ = kc.get_async("/legolas") or None, None
+
+    stored_value = ""
+    if value:
+        stored_value = value.get()
+    if stored_value:
+        assert stored_value[0] == b"hobbits"
+        return
+
+    raise KeyError
+
+
+def srvr(host: str) -> Dict:
+    response = check_output(
```

Comment:
A simple comment here on what this is doing would be helpful

Reply:
Yup, this is obtuse for those unfamiliar with ZK. I'll comment.